Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM perl:5.22
+
+ENV PERL_MM_USE_DEFAULT=1 PERL_CARTON_PATH=/carton
+
+COPY cpanfile cpanfile.snapshot /metacpan-web/
+WORKDIR /metacpan-web
+
+RUN cpanm App::cpm Carton && \
+    useradd -m metacpan-web -g users && \
+    mkdir /carton && \
+    cpm install -L /carton
+
+COPY . /metacpan-web
```

Comment:
Thanks, that's a good question.  In Docker Compose, most environments `COPY` the repo into the resulting image so that is can remain standalone, e.g. it can be put into hub.docker.com or local private Docker repository (like quay.io or gcr.io) and be able to pull/run from there without needing to clone the repo prior, e.g:

    docker pull my-private-docker-repo.io/metacpan-web:1.0
    docker run --rm -p 5001:5001 -it my-private-docker-repo.io/metacpan-web:1.0

In https://github.com/metacpan/metacpan-docker/pull/11 we do mount the repo into the container (via `services.web.volumes` in the Compose configuration,) so the `COPY` can be removed if we prefer to have a leaner "barebones" image of `metacpan-web` instead.  This would mean however that on ordinary `docker` runs we'd have to mount in the repo manually, e.g.:

    docker run -v /path/to/metacpan-web:/metacpan-web -it metacpan-web